test/pools/disjoint_pool.cpp

```diff
@@ -58,8 +58,13 @@ TEST_F(test, freeErrorPropagation) {
     void *ptr = umfPoolMalloc(pool, size);
 
     freeReturn = UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC;
```

Comment:
this `freeReturn` name is confusing, perhaps rename to `expectedRet` or something...?

Reply:
I think it is OK, take a look 30 lines above:
```c
        umf_result_t free(void *ptr, [[maybe_unused]] size_t size) noexcept {
            if (freeReturn == UMF_RESULT_SUCCESS) {
                ::free(ptr);
            }
            return freeReturn;
        }
```

Reply:
I'm not saying it's wrong, I'm just suggesting to rename the var. Now I see it's defined waay above, so perhaps we could rename it as a separate change/commit... but it'd be nice

Reply:
I meant this variable holds a value that **free()** will **return**, so I think `freeReturn` is a good name.

Reply:
This rename should apply to the whole file, but this PR doesn't touch any other line and still keeps the convention.

Reply:
> I'm not saying it's wrong, I'm just suggesting to rename the var. Now I see it's defined waay above, so perhaps we could rename it as a separate change/commit... but it'd be nice

@lukaszstolarczuk BTW this PR regards fixing memory leaks. If you want to rename a variable, you can submit a pull request.

Reply:
> I meant this variable holds a value that **free()** will **return**, so I think `freeReturn` is a good name.

Confusing for me was the usage in this part:

`EXPECT_EQ(freeRet, freeReturn);` 😉 

but, ok, I agree, it can be done as a separate change